.github/workflows/android-device-test/download-benchmark-results.mjs

```diff
@@ -0,0 +1,113 @@
+// This script parses logcat output from AWS Device Farm to extract benchmark results.
+// It also uploads the results to S3.
+// android-benchmark-render/{gitRevision}/{deviceModel}.json
+
+import {
+  DeviceFarmClient,
+  ListJobsCommand,
+  ListArtifactsCommand,
+} from "@aws-sdk/client-device-farm";
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
+
+import https from "node:https";
+import readline from "node:readline";
+
+const deviceFarmClient = new DeviceFarmClient({ region: "us-west-2" });
+
+function usage() {
+  console.error("Downloads benchmark results from AWS Device Farm");
+  console.error(`Usage: node ${process.argv.at(1)} RUN_ARN`);
+  process.exit(1);
+}
+
+if (process.argv.length !== 3) usage();
+
+const arn = process.argv.at(2);
+
+const listJobsCommand = new ListJobsCommand({ arn });
+const { jobs } = await deviceFarmClient.send(listJobsCommand);
+
+const passedJobs = jobs.filter((job) => job.result === "PASSED" && job.arn);
+const failedJobs = jobs.filter((job) => job.result !== "PASSED" || !job.arn);
+
+failedJobs.forEach((job) => {
+  console.error(
+    `job.result not PASSED. Skipping.\n$${JSON.stringify(job, null, 0)}`
+  );
+});
+
+const artifacts = (
+  await Promise.all(
+    passedJobs.map((job) =>
+      deviceFarmClient.send(
+        new ListArtifactsCommand({
+          arn: job.arn,
+          type: "FILE",
+        })
+      )
+    )
+  )
+).flatMap((listArtifactsOutput) =>
+  listArtifactsOutput.artifacts.filter(
+    (artifact) => artifact.type === "DEVICE_LOG" && artifact.name === "Logcat"
+  )
+);
+
+/**
+ * @param {string} artifactUrl 
+ * @returns {Promise<string>}
+ */
+function getBenchmarkResult(artifactUrl) {
```

Comment:
Consider moving this method to either the beginning of the file or the end, as it complicates reading the execution flow.
Also consider making more of this kind of method to better explain what the script is doing in every step.

Reply:
I only needed that function to promisify the Node.js API.

When I want to reuse a piece of code I'll refactor it into more functions later.